src/approves/identify-approvers.ts

```diff
@@ -1,12 +1,109 @@
-/*
-Function that returns required approvers based on file-changes.
-Probably logic from getReviewersBasedOnRule should be reused.
+import { info } from '../logger';
+import { Config, DefaultRules, Rule } from '../config/typings';
 
-But getReviewersBasedOnRule function returns only some specific random users from each group.
-And we need full object from our config.
- */
+function getReviewers({
+  reviewers,
+  createdBy,
+}: Pick<Rule, 'assign' | 'reviewers'> & {
+  createdBy: string;
+}): Set<string> {
+  const result = new Set<string>();
+  reviewers.forEach((reviewer) => {
+    if (reviewer === createdBy) {
+      return;
+    }
+    return result.add(reviewer);
+  });
+  return result;
+}
+
+function identifyReviewersByDefaultRules({
+  byFileGroups,
+  fileChangesGroups,
+  createdBy,
+}: {
+  byFileGroups: DefaultRules['byFileGroups'];
+  fileChangesGroups: string[];
+  requestedReviewerLogins: string[];
+  createdBy: string;
+}): Rule[] {
+  const rulesByFileGroup = byFileGroups;
+  const set: Rule[] = [];
+  fileChangesGroups.forEach((fileGroup) => {
+    const rules = rulesByFileGroup[fileGroup];
+    if (!rules) {
+      return;
+    }
+    rules.forEach((rule) => {
+      const reviewers = getReviewers({
+        reviewers: rule.reviewers,
+        createdBy,
+      });
+
+      set.push({
+        reviewers: [...reviewers],
+        required: rule.required,
+        assign: rule.assign,
+      });
+    });
+  });
+  return [...set];
```

Comment:
you can return `set` without `...`. 
```
return ruleList;
```

Since you've change the data type to the array, the `ruleList` name of the variable is more appropriate.